coalib/results/Diff.py

```diff
@@ -278,18 +278,37 @@ def modified(self):
         return result
 
     @property
+    def modified(self):
+        """
+        Calculates the modified file, after applying the Diff to the original.
+
+        This property also adds linebreaks at the end of each line.
+        If no newline was present at the end of file before, this state will
+        be preserved, except if the last line is deleted.
+        """
+        return self._generate_linebreaks(self._raw_modified())
+
+    @property
     def unified_diff(self):
         """
         Generates a unified diff corresponding to this patch.
 
+        Each change will be displayed on its own line. Additionally, the
+        unified diff preserves the EOF-state of the original file. This
+        means that the ``Diff`` will only have a linebreak on the last line,
+        if that was also present in the original file.
+
         Note that the unified diff is not deterministic and thus not suitable
         for equality comparison.
         """
-        return ''.join(difflib.unified_diff(
-            self.original,
-            self.modified,
+
+        list_unified_diff = list(difflib.unified_diff(
+            self._file,
+            self._raw_modified(),
             tofile=self.rename if isinstance(self.rename, str) else ''))
 
+        return ''.join(self._generate_linebreaks(list_unified_diff))
```

Comment:
Nah, it isn't fixed...yet.
The linebreaks are a first step, but the diff still needs filenames for it to work D:
It would be best if we could access the filename of the original, although a placeholder would work too.
I guess that should really be in a new issue, though.

Still not sure the linebreak addition should be moved though, cause that would require me to change all related tests again ;(